tests/integration/test_runtime_configurable_cache_size/test.py

```diff
@@ -101,9 +101,11 @@ def test_query_cache_size_is_runtime_configurable(start_cluster):
     node.query("SELECT 2 SETTINGS use_query_cache = 1, query_cache_ttl = 1")
     node.query("SELECT 3 SETTINGS use_query_cache = 1, query_cache_ttl = 1")
 
+    time.sleep(2)
```

Comment:
This is unreliable, it will remain flaky

Reply:
I actually break this test [here](https://s3.amazonaws.com/clickhouse-test-reports/0/8c6a6f2b71efe8500efd94de1177372e35bf1ae5/integration_tests__release__[1_4].html) in the last assert, sleep is required as i understand
```python
        res = node.query_with_retry(
            "SELECT value FROM system.asynchronous_metrics WHERE metric = 'QueryCacheEntries'",
            check_callback=lambda result: result == "1\n",
        )
>       assert res == "1\n"
E       AssertionError: assert '2\n' == '1\n'
E         - 1
E         + 2

```

Reply:
I wrote this test originally and found this open PR by luck.

There were three recorded test failures in the past 6 months or so:
- https://s3.amazonaws.com/clickhouse-test-reports/57933/ca3ca8ffe403a26a1ff5bc77adb8895c89b16970/integration_tests__tsan__[1_6].html
- https://s3.amazonaws.com/clickhouse-test-reports/60211/c5d3aef83e287b477f67bb5b0e97ed534b567270/integration_tests__asan__analyzer__[1_6].html
- https://s3.amazonaws.com/clickhouse-test-reports/60497/cb8d8594533de3773ad1f30f7a5792f1e7e1bb39/integration_tests__tsan__[1_6].html

which is super sporadic given how many CI tests run per day.

In all cases, we ran three queries with query cache TTL = 1 sec, then wait 2 sec, and find that the query cache contains only 1 entry whereas 2 entries are expected. I can honestly not think of a scenario where this can happen except one: the *asynchronous* metrics are updated in a weird time interval that they report only 1 entry. Instead of adding retries (l. 105, please revert all retries from this file which only hide problems), please add `SYSTEM RELOAD ASYNCHRONOUS METRICS` before l. 106, see https://github.com/ClickHouse/ClickHouse/pull/53710. When I am back from vacation, I can add some documentation for SYSTEM RELOAD ASYNCHRONOUS METRICS.